src/Iot.Device.Common/WeatherHelper.cs

```diff
@@ -0,0 +1,336 @@
+        /// From https://de.wikibooks.org/wiki/Tabellensammlung_Chemie/_Stoffdaten_Wasser, after D. Sonntag (1982)
+        /// </remarks>
+        public static Pressure CalculateSaturatedVaporPressureOverWater(Temperature airTemperature)
+        {
+            double tk = airTemperature.Kelvins;
+            double e_w = Math.Exp((-6094.4642 / tk) + 21.1249952 - (2.7245552E-2 * tk) + (1.6853396E-5 * tk * tk) +
+                                   (2.4575506 * Math.Log(tk)));
+            return Pressure.FromPascals(e_w);
+        }
+
+        /// <summary>
+        /// Calculates the saturated vapor pressure for a given air temperature over ice.
+        /// The formula used is valid for temperatures between -100°C and +0°C.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <returns>The saturated vapor pressure</returns>
+        /// <remarks>
+        /// From https://de.wikibooks.org/wiki/Tabellensammlung_Chemie/_Stoffdaten_Wasser, after D. Sonntag (1982)
+        /// </remarks>
+        public static Pressure CalculateSaturatedVaporPressureOverIce(Temperature airTemperature)
+        {
+            double tk = airTemperature.Kelvins;
+            double e_i = Math.Exp((-5504.4088 / tk) - 3.574628 - (1.7337458E-2 * tk) + (6.5204209E-6 * tk * tk) +
+                                  (6.1295027 * Math.Log(tk)));
+            return Pressure.FromPascals(e_i * 1.0041); // The table values are corrected by this
+        }
+
+        /// <summary>
+        /// Calculates the actual vapor pressure.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH)</param>
+        /// <returns>The actual vapor pressure</returns>
+        public static Pressure CalculateActualVaporPressure(Temperature airTemperature, RelativeHumidity relativeHumidity) =>
+            Pressure.FromHectopascals((relativeHumidity.Percent / 100.0 * CalculateSaturatedVaporPressureOverWater(airTemperature).Hectopascals));
+
+        /// <summary>
+        /// Calculates the dew point.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH)</param>
+        /// <returns>The dew point</returns>
+        /// <remarks>
+        /// Source https://en.wikipedia.org/wiki/Dew_point
+        /// </remarks>
+        public static Temperature CalculateDewPoint(Temperature airTemperature, RelativeHumidity relativeHumidity)
+        {
+            double pa = CalculateActualVaporPressure(airTemperature, relativeHumidity).Hectopascals;
+            double a = 6.1121; // hPa
+            double c = 257.14; // °C
+            double b = 18.678;
+            double dewPoint = (c * Math.Log(pa / a)) / (b - Math.Log(pa / a));
+            return Temperature.FromDegreesCelsius(dewPoint);
+        }
+
+        /*
```

Comment:
Maybe just add a note why it's commented, with a TODO: once density will be present, uncomment this part

Reply:
Great point. added.